### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by c00p75**
# integration specs for views and fix n+1 problems

- Secured app from n+1 problems
- Written integration tests with Capybara gem.
  - Add a test for the user's home page
  - Add tests for the user's show page
  - Add tests for the user posts page
  - Add tests for the post show page
  
  ## Screenshot 1
Screenshot 2023-06-20 094654

## Screenshot 2
Screenshot 2023-06-20 100315

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @posts = @user.posts.includes(:author)
   end
```
Comment (Owusu-Desmond) [inline app/controllers/posts_controller.rb]: - [ ] Please from the project requirement, you should make sure that the N+1 problem is solved when fetching all posts and their comments for a user so I think you should use `@posts = @user.posts.includes(:comments)` :pray:

File: spec/intergration/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    it "shows the user's first 3 posts" do
+      Post.create(author: @user, title: 'post 2', text: 'content', comments_counter: 1, likes_counter: 1)
+      Post.create(author: @user, title: 'post 3', text: 'content', comments_counter: 1, likes_counter: 1)
+      Post.create(author: @user, title: 'post 4', text: 'content', comments_counter: 1, likes_counter: 1)
+      visit user_path(@user)
+      expect(page.find_all('li h2').length).to eq(3)
+    end
+
+    it "shows a button to see all of the user's posts" do
+      expect(page.find(:xpath, '//button[1]').text).to eq('See all posts')
```
Comment (Owusu-Desmond) [inline spec/intergration/users_spec.rb]: - [ ] Please I suggest that you should have a Spec file that only tests the User index page and  a Spec file that only tests the User show page :pray:

File: spec/intergration/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    it 'show how many likes the post has' do
+      expect(page).to have_content('Likes: 1')
+    end
+
+    it 'show the post body' do
+      expect(page).to have_content('post 1')
+    end
+
+    it 'show the username of each commentor' do
+      expect(page).to have_content('Lee harmer')
```
Comment (Owusu-Desmond) [inline spec/intergration/posts_spec.rb]: - [ ] Please I suggest that you should have a Spec file that only tests the User post index page and  a Spec file that only tests the User post-show page :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @c00p75 and @calebchris000 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**